packages/matter.js/src/codec/BtpCodec.ts

```diff
@@ -63,7 +73,7 @@ export class BtpCodec {
         return this.decodeHandshakeRequestPayload(reader);
     }
 
-    static decodeBtpPacket(data: ByteArray): BtpPacket {
+    static decodeBtpPacket(data: ByteArray): DecodedBtpPacket {
         const reader = new DataReader(data, Endian.Little);
 
         const header = this.decodeBtpPacketHeader(reader);
```

Comment:
Similarly to encode, we should also add checks in decodePacket for the isContinuingSegment